### ISSUE TITLE
SQL: selecting a literal and an aggregate generates an error

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#41411]
```
sql> **select 123, max(salary) from test_emp group by gender;**
Server error [Server encountered an error [Cannot find group [123]]. [SqlIllegalArgumentException[Cannot find group [123]]
        at org.elasticsearch.xpack.sql.util.Check.notNull(Check.java:36)
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldAggregate.rule(QueryFolder.java:371)
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldAggregate.rule(QueryFolder.java:205)
        at org.elasticsearch.xpack.sql.tree.Node.lambda$transformUp$11(Node.java:196)
        at org.elasticsearch.xpack.sql.tree.Node.transformUp(Node.java:190)
        at org.elasticsearch.xpack.sql.tree.Node.transformUp(Node.java:196)
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldingRule.apply(QueryFolder.java:593)
```

**Comment by PhaedrusTheGreek** [comment]
I'm hitting this too using date functions with aggregates:

```
SELECT CURRENT_TIMESTAMP as result, stream, start_event, end_event, (end_event-start_event)  as duration from (SELECT stream, min(time) AS start_event, max(time) AS end_event FROM test group by stream)
```

fails with :

```
[essql] > Unexpected error from Elasticsearch: [sql_illegal_argument_exception] Cannot find group [result]
```

The `INTERVAL` function is also failing with a similar error

**Comment by matriv** [comment]
This is resolved probably as part of #49570
